### ISSUE TITLE
CupertinoTextField shows decimal point key even after setting decimal flag to false in TextInputType.numberWithOptions

### ISSUE THREAD

**Issue by vanelizarov** [flutter/flutter#27084]
## Steps to Reproduce

Issue's title speaks to itself: `CupertinoTextField` shows decimal point key even after setting `decimal` flag to `false` in `TextInputType.numberWithOptions`. And I need to be able to input only integer numbers.

Here's the piece of code, in which problem occurs:

```dart
import 'package:flutter/cupertino.dart';

void main() {
  runApp(App());
}

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return CupertinoApp(
      home: CupertinoPageScaffold(
        backgroundColor: Color(0xfff2f2f2),
        navigationBar: CupertinoNavigationBar(
          backgroundColor: Color(0xffffffff),
          actionsForegroundColor: Color(0xff000000),
        ),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: <Widget>[
            /// title
            Container(
              margin: const EdgeInsets.only(
                top: 32.0,
                bottom: 6.0,
              ),
              padding: const EdgeInsets.only(left: 16.0),
              child: Text(
                'input numeric value'.toUpperCase(),
                style: TextStyle(
                  color: const Color.fromRGBO(0, 0, 0, .5),
                  fontSize: 13.0,
                ),
              ),
            ),

            /// input
            Container(
              height: 44.0,
              child: CupertinoTextField(
                padding: const EdgeInsets.symmetric(
                  horizontal: 16.0,
                  vertical: 12.0,
                ),
                decoration: BoxDecoration(
                  color: Color(0xffffffff),
                ),
                style: TextStyle(
                  color: Color(0xff000000),
                  fontFamily: 'Corpid',
                  fontSize: 17.0,
                ),
                keyboardType: TextInputType.numberWithOptions(
                  decimal: false, // here it goes
                  signed: false,
                ),
              ),
            )
          ],
        ),
      ),
    );
  }
}

```

![image](https://user-images.githubusercontent.com/5096148/51748274-f94ca100-20bc-11e9-8021-13d7fe3c9d37.png)

## Logs

```
[✓] Flutter (Channel dev, v1.1.9, on Mac OS X 10.14.3 18D32a, locale en-RU)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
[✓] Android Studio (version 3.2)
[✓] VS Code (version 1.30.2)
[✓] Connected device (2 available)

• No issues found!
```

**Comment by zoechi** [comment]
Can you please check if this still happens in `master` channel.
This should have been fixed recently #25454

```
flutter channel master
flutter doctor
```

**Comment by zoechi** [comment]
It was a fix in the engine which usually takes a bit longer to land and therefore not in `dev` yet.

Duplicate of #25454
